src/gui/Measurement.cc

```diff
@@ -50,11 +50,43 @@ void Measurement::startMeasureAngle(void)
 }
 QString Measurement::statemachine(QPoint mouse) 
 {
-  if(qglview->measure_state == MEASURE_IDLE) return "";
+  if(qglview->measure_state == MEASURE_IDLE) return {};
   qglview->selectPoint(mouse.x(),mouse.y());
-  double ang=NAN;
   double dist=NAN;
   SelectedObject obj1, obj2, obj3;
+  SelectedObject ruler = {
+	  .type=SelectionType::SELECTION_INVALID
+  };
+  auto display_angle = [this] (Vector3d p1, Vector3d p2, Vector3d p3, Vector3d p4) 
+  {	
+	SelectedObject ruler;
+	Vector3d side1, side2;
+  	ruler = {
+            .type = SelectionType::SELECTION_SEGMENT,
+            .p1 = p1,
+            .p2 = p2
+        };
+	this->qglview->selected_obj.push_back(ruler);
+	ruler = {
+            .type = SelectionType::SELECTION_SEGMENT,
+            .p1 = p3,
+            .p2 = p4
+        };
+	this->qglview->selected_obj.push_back(ruler);
+
+        side1=(p2-p1).normalized();
+        side2=(p4-p3).normalized();
+        double ang=acos(side1.dot(side2))*180.0/G_PI;
+        if(!std::isnan(ang))
+        {
+          return QString("Angle  is %1 Degrees").arg(ang);
+        }
+        qglview->selected_obj.clear();
+        qglview->shown_obj = nullptr;
+        qglview->update();
+        qglview->measure_state = MEASURE_IDLE;
```

Comment:
This is just a nit, but it's sometimes more readable to handle the error inside the if statement and let the natural return path of the function be the good value.